src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExplicitConversion.cs

```diff
@@ -132,11 +132,14 @@ public bool Bind()
 
                 // But it failed, and so we know that the constant is not in range
 
+                Debug.Assert(
+                    _typeDest.GetTypeKind() != TypeKind.TK_MethodGroupType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_ErrorType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_ArgumentListType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_ParameterModifierType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_NullableType);
```

Comment:
That is a good point. It should at least be positive rather than negative. The cases where values are checked for being defined in an enum are positive. The case of `Debug.Assert(_typeDest.GetTypeKind() != TypeKind.TK_MethodGroupType);` in ImplicitConversion.cs is negative but specifically targetted. Here it's negative without a particularly good reason, so it's the inferior approach.